appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -71,14 +71,24 @@ export class AppSettingsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
     public async editSettingItem(oldKey: string, newKey: string, value: string): Promise<void> {
         const settings: StringDictionary = await this.ensureSettings();
```

Comment:
```
// clone cached settings into new object
const settings: StringDictionary = { ...await this.ensureSettings() };
```

Reply:
This is a great idea, but apparently only makes a shallow copy so anytime properties were referenced, it was still referring to the `this._settings.properties`.  I used the JSON method to make a deep-copy

Ref: https://scotch.io/bar-talk/copying-objects-in-javascript